### PR TITLE
support set engine-lable using command arguments

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:
This is part of https://github.com/pingcap/tiflash/pull/4805.
We need to change tiflash label to `tiflash_compute` instead of `tiflash` when tiflash is disaggregated.

### What is changed and how it works?
If user specifies `engine-label` in command line, we use it. If not, we use env variable of `engine-label` setted during compilation.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by guo-shaoge** [comment]
@CalvinNeo All CI passed, ptal again~

File: components/proxy_server/src/config.rs [hunk 1]
```diff
@@ -256,18 +256,14 @@ pub fn setup_default_tikv_config(default: &mut TikvConfig) {
...
-        Some(name) => name.to_owned(),
-    };
     config
         .server
         .labels
-        .insert(DEFAULT_ENGINE_LABEL_KEY.to_owned(), engine_name);
+        .entry(DEFAULT_ENGINE_LABEL_KEY.to_owned())
+        .or_insert(String::from(option_env!("ENGINE_LABEL_VALUE").unwrap()));
+
     config.raft_store.region_worker_tick_interval =
```
Comment (CalvinNeo) [inline components/proxy_server/src/config.rs:267]: Can we add a log here about `config.server.labels`'s final value?

Comment (guo-shaoge) [reply components/proxy_server/src/config.rs:271]: Added

**Review by CalvinNeo — APPROVED** [review]
LGTM, and please move the tests into a new function.

**Comment by guo-shaoge** [comment]
Don't know why this case failed: thread 'cases::test_merge::test_node_merge_restart' panicked at 'can't get value None for key 6B31', components/test_raftstore/src/util.rs:75:5. https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb-engine-ext%2Fghpr_unit_test/detail/ghpr_unit_test/130/pipeline

**Comment by CalvinNeo** [comment]
> Don't know why this case failed: thread 'cases::test_merge::test_node_merge_restart' panicked at 'can't get value None for key 6B31', components/test_raftstore/src/util.rs:75:5. https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb-engine-ext%2Fghpr_unit_test/detail/ghpr_unit_test/130/pipeline

These tests are adopt from TiKV. They occasionally fails. Just rerun it.

**Pull request merged**